openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -157,301 +63,54 @@ def override_for_request(self, switch_name, active=True):
-    return set(waffle_flag_custom_attributes)
-
-# .. toggle_name: WAFFLE_FLAG_CUSTOM_ATTRIBUTES
-# .. toggle_implementation: DjangoSetting
-# .. toggle_default: False
-# .. toggle_description: A list of waffle flags to track with custom attributes having
-#   values of (True, False, or Both).
-# .. toggle_use_cases: opt_in
-# .. toggle_creation_date: 2020-06-17
-# .. toggle_warnings: Intent is for temporary research (1 day - several weeks) of a flag's usage.
-_WAFFLE_FLAG_CUSTOM_ATTRIBUTES = 'WAFFLE_FLAG_CUSTOM_ATTRIBUTES'
-
-# set of waffle flags that should be instrumented with custom attributes
-_WAFFLE_FLAG_CUSTOM_ATTRIBUTE_SET = _get_waffle_flag_custom_attributes_set()
-
-
-class WaffleFlag(object):
-    """
-    Represents a single waffle flag, using a cached waffle namespace.
-    """
-    # use a WeakSet so these instances can be garbage collected if need be
-    _class_instances = WeakSet()
-
-    def __init__(self, waffle_namespace, flag_name, module_name=None):
-        """
-        Initializes the waffle flag instance.
-
-        Arguments:
-            waffle_namespace (WaffleFlagNamespace | String): Namespace for this flag.
-            flag_name (String): The name of the flag (without namespacing).
-            module_name (String): The name of the module where the flag is created. This should be ``__name__`` in most
-            cases.
-        """
-        if isinstance(waffle_namespace, six.string_types):
-            waffle_namespace = WaffleFlagNamespace(name=waffle_namespace)
-
-        self.waffle_namespace = waffle_namespace
-        self.waffle_namespace = waffle_namespace
-        self.flag_name = flag_name
-        self._module_name = module_name or ""
-        self._class_instances.add(self)
-
-    @classmethod
-    def get_instances(cls):
-        """ Returns a WeakSet of the instantiated instances of WaffleFlag. """
-        return cls._class_instances
-
-    @property
-    def module_name(self):
-        """
-        Returns the module name. This is cached to work with the WeakSet instances.
-        """
-        return self._module_name
-
-    @module_name.setter
-    def module_name(self, value):
-        self._module_name = value
-
-    @property
-    def namespaced_flag_name(self):
-        """
-        Returns the fully namespaced flag name.
-        """
-        return self.waffle_namespace._namespaced_name(self.flag_name)
-
-    def is_enabled(self):
-        """
-        Returns whether or not the flag is enabled.
-        """
-        return self.waffle_namespace.is_flag_active(self.flag_name)
-
-    @contextmanager
-    def override(self, active=True):
         # TODO We can move this import to the top of the file once this code is
         # not all contained within the __init__ module.
         from openedx.core.djangoapps.waffle_utils.testutils import override_waffle_flag
+
         with override_waffle_flag(self, active):
             yield
```

Comment:
> If we need this, can we now import override_waffle_flag from edx-toggles at the top of the file and remove the TODO?

No, I don't think so, because we don't want to import test modules in production.

> Note: if we decide to remove this and it makes sense in a follow-up PR, just let me know.

I honestly have no idea. Whatever we do with this, I suggest we do it in a separate PR because it's going to impact a lot of code.

Reply:
Ok.  Can you just add a task to come back to this?  Also, should we at least be switching to override_waffle_flag from edx-toggles, or do you want use of that to also be part of follow-up work?

Reply:
>  Can you just add a task to come back to this

:ok_hand: 

> Also, should we at least be switching to override_waffle_flag from edx-toggles, or do you want use of that to also be part of follow-up work?

I think this should be part of the waffle_utils cleanup as it will impact many files.